File: iRHSCBook/RHSCMemberDetailViewController.m
//
//  RHSCMemberDetailViewController.m
//  iRHSCBook
//
//  Created by Bruce Hunter on 2014-07-08.
//  Copyright (c) 2014 Richmond Hill Squash Club. All rights reserved.
//

#import "RHSCMemberDetailViewController.h"
#import <MessageUI/MessageUI.h>

@interface RHSCMemberDetailViewController ()

@end

@implementation RHSCMemberDetailViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.navigationItem.title = [NSString stringWithFormat:@"%@ %@",_member.firstName,_member.lastName];
    self.emailLabel.text = @"";
    self.emailBtn.hidden = YES;
    if (_member.email != (NSString *)[NSNull null]) {
        if (![_member.email isEqualToString:@"NULL"]) {
            self.emailLabel.text = _member.email;
            self.emailBtn.hidden = NO;
        }
    }
    self.phone1Label.text = @"";
    self.ph1SMSBtn.hidden = YES;
    self.ph1CallBtn.hidden = YES;
    if (_member.phone1 != (NSString *)[NSNull null]) {
        if (![_member.phone1 isEqualToString:@"NULL"]) {
            self.phone1Label.text = _member.phone1;
            self.ph1SMSBtn.hidden = NO;
            self.ph1CallBtn.hidden = NO;
        }
    }
    self.phone2Label.text = @"";
    self.ph2SMSBtn.hidden = YES;
    self.ph2CallBtn.hidden = YES;
    if (_member.phone2 != (NSString *)[NSNull null]) {
        if (![_member.phone2 isEqualToString:@"NULL"]) {
            self.phone2Label.text = _member.phone2;
            self.ph2SMSBtn.hidden = NO;
            self.ph2CallBtn.hidden = NO;
        }
    }
}

- (IBAction)emailMember:(id)sender {
    if ([MFMailComposeViewController canSendMail])
    {
        // Email Subject
        NSString *emailTitle = @"";
        // Email Content
        NSString *messageBody = @"";
        // To address
        NSArray *toRecipents = [NSArray arrayWithObject:[self.member email]];
        
        MFMailComposeViewController *mc = [[MFMailComposeViewController alloc] init];
        mc.mailComposeDelegate = self;
        [mc setSubject:emailTitle];
        [mc setMessageBody:messageBody isHTML:NO];
        [mc setToRecipients:toRecipents];
        
        // Present mail view controller on screen
        [self presentViewController:mc animated:YES completion:NULL];
    } else {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Cannot send email"
                                                        message:@"Cannot email from this device"
                                                       delegate:nil
                                              cancelButtonTitle:@"OK"
                                              otherButtonTitles:nil];
        [alert show];
    }
}

- (void) mailComposeController:(MFMailComposeViewController *)controller didFinishWithResult:(MFMailComposeResult)result error:(NSError *)error
{
    switch (result)
    {
        case MFMailComposeResultCancelled:
            NSLog(@"Mail cancelled");
            break;
        case MFMailComposeResultSaved:
            NSLog(@"Mail saved");
            break;
        case MFMailComposeResultSent:
            NSLog(@"Mail sent");
            break;
        case MFMailComposeResultFailed:
            NSLog(@"Mail sent failure: %@", [error localizedDescription]);
            break;
        default:
            break;
    }
    
    // Close the Mail Interface
    [self dismissViewControllerAnimated:YES completion:NULL];
}

- (IBAction)phone1Member:(id)sender {
    NSLog(@"Phoning using number 1");
    NSString *cleanedString = [[[self.member phone1] componentsSeparatedByCharactersInSet:[[NSCharacterSet characterSetWithCharactersInString:@"0123456789-+()"] invertedSet]] componentsJoinedByString:@""];
    NSString *phoneNumber = [@"telprompt://" stringByAppendingString:cleanedString];
    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:phoneNumber]]) {
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:phoneNumber]];
    } else {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Cannot place call"
                                                        message:@"Cannot phone from this device"
                                                       delegate:nil
                                              cancelButtonTitle:@"OK"
                                              otherButtonTitles:nil];
        [alert show];
        
    }
}

- (IBAction)phone2Member:(id)sender {
    NSLog(@"Phoning using number 2");
    NSString *cleanedString = [[[self.member phone2] componentsSeparatedByCharactersInSet:[[NSCharacterSet characterSetWithCharactersInString:@"0123456789-+()"] invertedSet]] componentsJoinedByString:@""];
    NSString *phoneNumber = [@"telprompt://" stringByAppendingString:cleanedString];
    if ([[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:phoneNumber]]) {
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:phoneNumber]];
    } else {
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Cannot place call"
                                                        message:@"Cannot phone from this device"
                                                       delegate:nil
                                              cancelButtonTitle:@"OK"
                                              otherButtonTitles:nil];
        [alert show];
        
    }
}

- (IBAction)sms1Member:(id)sender {
	MFMessageComposeViewController *controller = [[MFMessageComposeViewController alloc] init];
    NSLog(@"Sending SMS using number 1");
    NSString *cleanedString = [[[self.member phone1] componentsSeparatedByCharactersInSet:[[NSCharacterSet characterSetWithCharactersInString:@"0123456789-+()"] invertedSet]] componentsJoinedByString:@""];
	if([MFMessageComposeViewController canSendText])
	{
		controller.body = @"";
		controller.recipients = [NSArray arrayWithObject:cleanedString];
		controller.messageComposeDelegate = self;
        [self presentViewController:controller animated:YES completion:NULL];
	}
}

- (IBAction)sms2Member:(id)sender {
	MFMessageComposeViewController *controller = [[MFMessageComposeViewController alloc] init];
    NSLog(@"Sending SMS using number 2");
    NSString *cleanedString = [[[self.member phone2] componentsSeparatedByCharactersInSet:[[NSCharacterSet characterSetWithCharactersInString:@"0123456789-+()"] invertedSet]] componentsJoinedByString:@""];
	if([MFMessageComposeViewController canSendText])
	{
		controller.body = @"";
		controller.recipients = [NSArray arrayWithObject:cleanedString];
		controller.messageComposeDelegate = self;
        [self presentViewController:controller animated:YES completion:NULL];
	}
}

- (void)messageComposeViewController:(MFMessageComposeViewController *)controller didFinishWithResult:(MessageComposeResult)result
{
	switch (result) {
		case MessageComposeResultCancelled:
//			NSLog(@"Cancelled");
			break;
		case MessageComposeResultFailed: {
            UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Cannot send message"
                                                            message:@"Cannot SMS from this device"
                                                           delegate:nil
                                                  cancelButtonTitle:@"OK"
                                                  otherButtonTitles:nil];
            [alert show];
			break;
        }
		case MessageComposeResultSent:
			break;
		default:
			break;
	}
    [self dismissViewControllerAnimated:YES completion:NULL];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: iRHSCBook/RHSCMemberDetailViewController.swift
//
//  RHSCMemberDetailViewController.swift
//  iRHSCBook
//
//  Created by Bruce Hunter on 2015-10-27.
//  Copyright © 2015 Richmond Hill Squash Club. All rights reserved.
//

import Foundation
import MessageUI

class RHSCMemberDetailViewController : UIViewController,MFMailComposeViewControllerDelegate,MFMessageComposeViewControllerDelegate {
    
    @IBOutlet weak var emailLabel : UILabel? = nil
    @IBOutlet weak var phone1Label : UILabel? = nil
    @IBOutlet weak var phone2Label : UILabel? = nil
    @IBOutlet weak var emailBtn : UIButton? = nil
    @IBOutlet weak var ph1SMSBtn : UIButton? = nil
    @IBOutlet weak var ph1CallBtn : UIButton? = nil
    @IBOutlet weak var ph2SMSBtn : UIButton? = nil
    @IBOutlet weak var ph2CallBtn : UIButton? = nil
    
    var errorAlert : UIAlertController? = nil
    
    var member : RHSCMember? = nil
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.navigationItem.title = String.init(format: "%@ %@", arguments: [member!.firstName!,member!.lastName!])
        
        self.emailLabel!.text = ""
        self.emailBtn!.hidden = true
        if ((member!.email) != nil) {
            if (member!.email! != "NULL") {
                self.emailLabel!.text = member!.email!
                self.emailBtn!.hidden = false
            }
        }
        self.phone1Label!.text = ""
        self.ph1SMSBtn!.hidden = true
        self.ph1CallBtn!.hidden = true
        if ((member!.phone1) != nil) {
            if (member!.phone1! != "NULL") {
                self.phone1Label!.text = member!.phone1!
                self.ph1SMSBtn!.hidden = false
                self.ph1CallBtn!.hidden = false
            }
        }
        self.phone2Label!.text = ""
        self.ph2SMSBtn!.hidden = true
        self.ph2CallBtn!.hidden = true
        if ((member!.phone2) != nil) {
            if (member!.phone2! != "NULL") {
                self.phone2Label!.text = member!.phone2!
                self.ph2SMSBtn!.hidden = false
                self.ph2CallBtn!.hidden = false
            }
        }
    }
    
    @IBAction func emailMember(sender: AnyObject) {
        if (MFMailComposeViewController.canSendMail()) {
            // Email Subject
            let emailTitle = ""
            // Email Content
            let messageBody = ""
            // To address
            let toRecipients = Array<String>(arrayLiteral: member!.email!)
    
            let mc = MFMailComposeViewController()
            mc.mailComposeDelegate = self
            mc.setSubject(emailTitle)
            mc.setMessageBody(messageBody, isHTML: false)
            mc.setToRecipients(toRecipients)
    
            // Present mail view controller on screen
            self.presentViewController(mc, animated: true, completion: nil)
        } else {
            self.errorAlert = UIAlertController(title: "Unavailable",
                message: "Cannot email from this device", preferredStyle: .Alert)
            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {
                // do some task
                dispatch_async(dispatch_get_main_queue(), {
                    self.presentViewController(self.errorAlert!, animated: true, completion: nil)
                    let delay = 2.0 * Double(NSEC_PER_SEC)
                    let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
                    dispatch_after(time, dispatch_get_main_queue(), {
                        self.errorAlert!.dismissViewControllerAnimated(true, completion: nil)
                    })
                })
            })
        }
    }
    
    func mailComposeController(controller: MFMailComposeViewController, didFinishWithResult result: MFMailComposeResult, error: NSError?) {
        switch (result) {
        case MFMailComposeResultCancelled:
            break;
        case MFMailComposeResultSaved:
            break;
        case MFMailComposeResultSent:
            break;
        case MFMailComposeResultFailed:
            break;
        default:
            break;
        }
    
        // Close the Mail Interface
        dismissViewControllerAnimated(true, completion: nil)
    }
    
    @IBAction func phone1Member(sender:AnyObject) {
        //    NSLog(@"Phoning using number 1");
        let cleanedString = member!.phone1!
            .componentsSeparatedByCharactersInSet(NSCharacterSet(charactersInString: "0123456789-+()")
                .invertedSet).joinWithSeparator("")
        let phoneNumber = String.init(format: "telprompt://%@", arguments: [cleanedString])
        if (UIApplication.sharedApplication().canOpenURL(NSURL(string:phoneNumber)!)) {
            UIApplication.sharedApplication().openURL(NSURL(string:phoneNumber)!)
        } else {
            self.errorAlert = UIAlertController(title: "Unavailable",
                message: "Cannot phone from this device", preferredStyle: .Alert)
            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {
                // do some task
                dispatch_async(dispatch_get_main_queue(), {
                    self.presentViewController(self.errorAlert!, animated: true, completion: nil)
                    let delay = 2.0 * Double(NSEC_PER_SEC)
                    let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
                    dispatch_after(time, dispatch_get_main_queue(), {
                        self.errorAlert!.dismissViewControllerAnimated(true, completion: nil)
                    })
                })
            })
        }
    }
    
    @IBAction func phone2Member(sender:AnyObject) {
        //    NSLog(@"Phoning using number 2");
        let cleanedString = member!.phone2!
            .componentsSeparatedByCharactersInSet(NSCharacterSet(charactersInString: "0123456789-+()")
                .invertedSet).joinWithSeparator("")
        let phoneNumber = String.init(format: "telprompt://%@", arguments: [cleanedString])
        if (UIApplication.sharedApplication().canOpenURL(NSURL(string:phoneNumber)!)) {
            UIApplication.sharedApplication().openURL(NSURL(string:phoneNumber)!)
        } else {
            self.errorAlert = UIAlertController(title: "Unavailable",
                message: "Cannot phone from this device", preferredStyle: .Alert)
            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {
                // do some task
                dispatch_async(dispatch_get_main_queue(), {
                    self.presentViewController(self.errorAlert!, animated: true, completion: nil)
                    let delay = 2.0 * Double(NSEC_PER_SEC)
                    let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
                    dispatch_after(time, dispatch_get_main_queue(), {
                        self.errorAlert!.dismissViewControllerAnimated(true, completion: nil)
                    })
                })
            })
        }
    }
    
    @IBAction func sms1Member(sender:AnyObject) {
        let controller = MFMessageComposeViewController()
        //    NSLog(@"Sending SMS using number 1");
        let cleanedString = member!.phone1!
            .componentsSeparatedByCharactersInSet(NSCharacterSet(charactersInString: "0123456789-+()")
                .invertedSet).joinWithSeparator("")
        if(MFMessageComposeViewController.canSendText()) {
            controller.body = ""
            controller.recipients = Array<String>([cleanedString])
            controller.messageComposeDelegate = self;
            self.presentViewController(controller, animated: true, completion: nil)
        }
    }
    
    @IBAction func sms2Member(sender:AnyObject) {
        let controller = MFMessageComposeViewController()
        //    NSLog(@"Sending SMS using number 1");
        let cleanedString = member!.phone2!
            .componentsSeparatedByCharactersInSet(NSCharacterSet(charactersInString: "0123456789-+()")
                .invertedSet).joinWithSeparator("")
        if(MFMessageComposeViewController.canSendText()) {
            controller.body = ""
            controller.recipients = Array<String>([cleanedString])
            controller.messageComposeDelegate = self;
            self.presentViewController(controller, animated: true, completion: nil)
        }
    }
    
    func messageComposeViewController(controller: MFMessageComposeViewController, didFinishWithResult result: MessageComposeResult) {
        switch (result) {
        case MessageComposeResultCancelled:
            //			NSLog(@"Cancelled");
            break;
        case MessageComposeResultFailed:
            self.errorAlert = UIAlertController(title: "Unavailable",
                message: "Cannot SMS from this device", preferredStyle: .Alert)
            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {
                // do some task
                dispatch_async(dispatch_get_main_queue(), {
                    self.presentViewController(self.errorAlert!, animated: true, completion: nil)
                    let delay = 2.0 * Double(NSEC_PER_SEC)
                    let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
                    dispatch_after(time, dispatch_get_main_queue(), {
                        self.errorAlert!.dismissViewControllerAnimated(true, completion: nil)
                    })
                })
            })
            break;
        case MessageComposeResultSent:
            break;
        default:
            break;
        }
        dismissViewControllerAnimated(true, completion: nil)
    }

}
